mithril-stm/src/membership_commitment/mod.rs

```diff
@@ -1,3 +1,23 @@
 mod merkle_tree;
 
 pub use merkle_tree::*;
+
+use blake2::{Blake2b, digest::consts::U32};
+use digest::{Digest, FixedOutput};
+use std::fmt::Debug;
+
+pub trait MembershipDigest {
```

Comment:
IMO this is not the best place for this trait as this module should be used only to implement the commitment schemes.